app/src/main/java/com/github/se/cyrcle/model/user/UserViewModel.kt

```diff
@@ -95,20 +111,35 @@ class UserViewModel(
   }
 
   /**
-   * Updates a user in the Firestore database.
+   * Updates a user in the Firestore database. If a user has Uri for the profile picture, it will
+   * upload the image to the cloud storage and then delete the Uri.
    *
    * @param user the user to update
    */
-  fun updateUser(user: User) {
-    userRepository.updateUser(
-        user,
-        onSuccess = { Log.d("UserViewModel", "User updated") },
-        onFailure = { exception ->
-          Log.e(
-              "com.github.se.cyrcle.model.user.UserViewModel",
-              "Failed to update user: ${user.public.userId}",
-              exception)
-        })
+  fun updateUser(user: User, context: Context? = null) {
+    if (user.public.userId != currentUser.value?.public?.userId) {
+      Log.e("UserViewModel", "Trying to update a user that is not the current user")
+      return
+    }
+
+    Log.d("UserViewModel", "Updating user: $user")
+    // First update the user object in the view model.
+    setCurrentUser(user)
+    // Then update the user object in the database
+    if (user.localSession?.profilePictureUri == null || context == null) {
+      userRepository.updateUser(
+          user,
+          onSuccess = { Log.d("UserViewModel", "User updated in the database") },
```

Comment:
For such a central function i would add an argument `OnSuccess` and put your log as default parameter so that the function stays as modular and flexible as it could be what do you think ?

Reply:
Sure, I can add one, I don't see an immediate reason for it but it doesn't hurt to but one and make the code more flexible